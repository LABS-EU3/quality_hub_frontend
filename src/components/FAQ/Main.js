import React from 'react';
import { Link, NavLink, Route } from 'react-router-dom';
import styled, { css } from 'styled-components';
import SearchIcon from '@material-ui/icons/Search';
import PersonIcon from '@material-ui/icons/Person';
import PaymentIcon from '@material-ui/icons/Payment';
import logo from '../../img/firelogo.png';
import General from './GeneralFaq';
import Profile from './ProfileFaq';
import Payment from './PaymentFaq';

const facebookIcon = require('../../img/fb.png');
const githubIcon = require('../../img/github-1.png');
const instagramIcon = require('../../img/instagram-logo-1.png');
const linkedInIcon = require('../../img/linkedin.png');
const twitterIcon = require('../../img/twitter-1.png');

// Nav Bar
const mobileHidden = css`
  @media only screen and (max-width: 600px) {
    display: none;
  }
`;

const NavbarContainer = styled.div`
  height: 4em;
  width: 100%;
  background: white;
  display: flex;
  justify-content: space-between;
  position: sticky;
  top: 0;
  z-index: 11;
  box-shadow: 0px 3px 4px -2px rgba(150, 150, 150, 1);

  @media only screen and (max-width: 600px) {
    justify-content: space-evenly;
  }
`;

const Logo = styled.div`
  width: 4.5em;
  height: 5em;
  background-image: url(${logo});
  background-repeat: no-repeat;
  transform: scale(0.7);
  ${mobileHidden};
`;

const LogoTitleContainer = styled.div`
  width: 30%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

const NavTitleContainer = styled.div`
  font-family: 'Ubuntu';
  font-style: normal;
  font-weight: normal;
  color: grey;
  font-size: 18px;

  .navLink {
    text-decoration: none;
    color: grey;
  }
`;

const LinksContainer = styled.div`
  width: 30%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding-right: 0.2em;
`;

const NavbarLink = styled.div`
  font-family: Ubuntu;
  font-size: 14px;
  color: black;
  ${mobileHidden};

  a {
    text-decoration: none;
    color: black;
  }
`;

export const buttonTheme = {
  text: 'white',
  background: '#408f53',
};

export const invertTheme = {
  text: '#4fad65',
  background: 'white',
};

// Main content

const MainContainer = styled.div`
  background: #f6f9fc;
  height: 150vh;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;

  h2 {
    margin-top: 50px;
    font-family: 'Ubuntu', sans-serif;
    color: #4fad65;
    font-size: 40px;
  }
`;

const CategoryContainer = styled.div`
  display: flex;
  margin-bottom: 20px;

  .styledBox {
    width: 12em;
    border-radius: 10px;
    height: 150px;
    margin: 3em;
    border-width: 10px;
    border: 2px solid grey;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-decoration: none;
    /* box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.15); */
    box-shadow: 0 6px 10px rgba(50, 50, 93, 0.1);

    h3 {
      text-align: center;
      color: grey;
    }
  }

  .activeClassNav {
    background: #1e3f1f;
    border: 2px solid white;

    h3 {
      color: white;
    }
  }
`;

const iconStyles = {
  fontSize: '3.5em',
  color: '#4fad65',
};

// Footer

const FooterConatiner = styled.div`
  height: 50vh;
  width: 100%;
  background: #292d38;
  p {
    font-family: Ubuntu;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 21px;
    text-align: center;
    color: #ffffff;
  }
`;

const FooterTopSection = styled.div`
  display: flex;
  justify-content: space-evenly;
`;

const LogoFooter = styled.div`
  margin-top: 1.5%;
  margin-left: 10%;
  height: 10em;
  width: 5em;
  background-image: url(${logo});
  background-repeat: no-repeat;
`;

const LogoTitleContainerFooter = styled.div`
  margin-top: 5%;
  width: 30%;
  display: flex;
  justify-content: flex-start;
`;

const FooterTextStyle = styled.div`
  margin-top: 2%;
  h3 {
    font-family: Ubuntu;
    font-style: normal;
    font-weight: bold;
    font-size: 12px;
    line-height: 21px;
    display: flex;
    align-items: center;
    text-align: center;
    color: #ffffff;
  }
  p {
    font-family: Ubuntu;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 21px;
    display: flex;
    align-items: center;
    text-align: center;
    color: #ffffff;
  }
`;

const FooterBottomSection = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Icons = styled.div`
  margin-right: 1.5%;
  height: 1.6em;
  img {
    max-height: 100%;
  }
`;

const MainFaq = () => {
  return (
    <MainContainer>
      <NavbarContainer>
        <LogoTitleContainer>
          <NavLink to='/'>
            <Logo />
          </NavLink>
          <NavTitleContainer>
            <NavLink className='navLink' to='/'>
              <h1>DevCoach</h1>
            </NavLink>
          </NavTitleContainer>
        </LogoTitleContainer>
        <LinksContainer>
          <NavbarLink>
            <a href='/dashboard'>DashBoard</a>
          </NavbarLink>
          <NavbarLink>
            <a href='/faq/general'>FAQ</a>
          </NavbarLink>
          <NavbarLink>
            <a href='#'>About</a>
          </NavbarLink>
          <NavbarLink>
            <a href='/login'>Login</a>
          </NavbarLink>
        </LinksContainer>
      </NavbarContainer>
      <h2>FAQs</h2>
      <CategoryContainer>
        <NavLink
          className='styledBox'
          activeClassName='activeClassNav'
          to='/faq/general'
        >
          <h3>General</h3>
          <SearchIcon style={iconStyles} />
        </NavLink>

        <NavLink
          className='styledBox'
          activeClassName='activeClassNav'
          to='/faq/profile'
        >
          <h3>Profile</h3>
          <PersonIcon style={iconStyles} />
        </NavLink>
        <NavLink
          className='styledBox'
          activeClassName='activeClassNav'
          to='/faq/payment'
        >
          <h3>Payment</h3>
          <PaymentIcon style={iconStyles} />
        </NavLink>
      </CategoryContainer>

      <Route exact path='/faq/general' component={General} />
      <Route exact path='/faq/profile' component={Profile} />
      <Route exact path='/faq/payment' component={Payment} />

      <FooterConatiner>
        <FooterTopSection>
          <LogoTitleContainerFooter>
            <LogoFooter />
            <NavTitleContainer>
              <h1>DevCoach</h1>
            </NavTitleContainer>
          </LogoTitleContainerFooter>
          <FooterTextStyle>
            <h3>SUPPORT</h3>
            <p>How it works</p>
          </FooterTextStyle>
          <FooterTextStyle>
            <h3>PRODUCT</h3>
            <p>Product feedback</p>
            <p>About the team</p>
          </FooterTextStyle>
          <FooterTextStyle>
            <h3>HELP</h3>
            <p>FAQ</p>
            <p>Contact us</p>
            <p>Career</p>
          </FooterTextStyle>
          <FooterTextStyle>
            <h3>LEGAL</h3>
            <p>Privacy Policy</p>
            <p>Terms and Conditions</p>
            <p>Manage Cookies</p>
          </FooterTextStyle>
        </FooterTopSection>
        <FooterBottomSection>
          <Icons>
            <img src={githubIcon} />
          </Icons>
          <Icons>
            <img src={facebookIcon} />
          </Icons>
          <Icons>
            <img src={instagramIcon} />
          </Icons>
          <Icons>
            <img src={linkedInIcon} />
          </Icons>
          <Icons>
            <img src={twitterIcon} />
          </Icons>
        </FooterBottomSection>
        <p>
          {' '}
          hello@dev-coach.com - copyright Â© Devcoach 2019. All rights
          reserved.
        </p>
      </FooterConatiner>
    </MainContainer>
  );
};

export default MainFaq;
