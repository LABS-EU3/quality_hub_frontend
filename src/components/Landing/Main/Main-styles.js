import styled from 'styled-components';
import devices from '../../devices';

export const MainContainer = styled.div`
  max-width: 100%;
  display: flex;
  flex-direction: column;
  margin-top: 10rem;

  .main-about-top {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    padding: 0 2rem;
    text-align: center;
  }

  .main-about-title {
    color: #141414;
    font-size: 2rem;
    font-weight: bold;
    margin-bottom: 2rem;
    line-height: 1.2;
  }

  .main-about-intro {
    text-align: center;
    font-size: 1.6rem;
    line-height: 1.3;
    color: #545454;
    font-weight: bold;
    max-width: 60rem;
  }

  .logo {
    color: #408f53;
  }

  .main-about-bottom {
    display: flex;
    flex-direction: column;
    justify-content: center;
    background: #292d38;
    margin: 10rem 0;
    padding: 3rem 1rem;

    h4 {
      font-size: 2rem;
      text-align: center;
      color: #efefef;
      margin: 0;
    }

    .about-cards {
      display: flex;
      justify-content: space-evenly;
      flex-wrap: wrap;
    }

    .about-card {
      border-radius: 6px;
      box-shadow: 0px 10px 10px #161616;
      background: #efefef;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: space-between;
      height: 25rem;
      max-width: 25rem;
      margin-top: 3rem;

      @media ${devices.mobile} {
        width: 80%;
      }

      .image {
        width: 100%;
      }

      img {
        border-radius: 6px 6px 0 0;
        width: 100%;
        height: 12rem;
        object-fit: cover;
      }

      .card-info {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: space-evenly;
        height: 100%;
      }

      h6 {
        margin: 0;
        font-weight: bold;
        font-size: 1.6rem;
      }

      p {
        font-size: 1rem;
        padding: 0 2.2rem;
        text-align: center;
        color: #545454;
        margin: 0;
        line-height: 1.3;
      }
    }
  }

  .user-stories-container {
    margin-bottom: 10rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    h2 {
      font-weight: bold;
      font-size: 2.2rem;
      margin-bottom: 5rem;
      max-width: 60rem;
      padding: 0 2rem;
      text-align: center;
      line-height: 1.3;
      color: #141414;

      @media ${devices.tablet} {
        margin-bottom: 2.5rem;
      }
    }

    .user-stories {
      display: flex;
      flex-wrap: wrap;
      justify-content: center;
      align-items: center;
      width: 100%;
    }

    .user-story {
      margin: 1.3rem 3rem;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: space-between;
      width: 28rem;

      p {
        line-height: 2;
        font-size: 1.1rem;
        color: #545454;

        @media ${devices.tablet} {
          text-align: center;
        }
      }
    }

    .user-header {
      width: 100%;
      display: flex;
      align-items: center;
      margin: 1rem 0;

      @media ${devices.tablet} {
        justify-content: center;
        flex-direction: column;
      }

      img {
        width: 5rem;
        height: 5rem;
        border-radius: 50%;
        object-fit: cover;
        margin-right: 1.2rem;

        @media ${devices.tablet} {
          margin: 0 0 1rem 0;
        }
      }

      h3 {
        font-weight: bold;
        font-size: 1.4rem;
        color: #141414;
        margin: 0;
      }
    }
  }
`;
