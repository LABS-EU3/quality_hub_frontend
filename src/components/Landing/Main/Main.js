import React from 'react';
import { MainContainer } from './Main-styles';

import aboutOne from './img/how-it-works-1.jpg';
import aboutTwo from './img/how-it-works-2.jpg';
import aboutThree from './img/how-it-works-3.jpg';
import personOne from './img/person-1.jpg';
import personTwo from './img/person-2.jpg';
import personThree from './img/person-3.jpg';

const Main = () => (
  <MainContainer>
    <div className='main-about-top'>
      <h2 className='main-about-title'>
        Enhance your skills. Land your dream job.
        <span className='logo'> DevCoach</span>
      </h2>
      <p className='main-about-intro'>
        We connect developers looking to improve their interviewing
        technique with experienced pros who have mastered the
        technical interview and can coach you through the process from
        start to finish.
      </p>
    </div>

    <div className='main-about-bottom'>
      <h4>How does DevCoach work?</h4>
      <div className='about-cards'>
        <div className='about-card'>
          <div className='image'>
            <img src={aboutTwo} alt='hand shake´' />
          </div>
          <div className='card-info'>
            <h6>Find Your Coach</h6>
            <p>
              Match with experienced professionals tailored to you
            </p>
          </div>
        </div>
        <div className='about-card'>
          <div className='image'>
            <img src={aboutOne} alt='analytics´' />
          </div>
          <div className='card-info'>
            <h6>Enhance Your Skills</h6>
            <p>Practice mock interviews one on one with your coach</p>
          </div>
        </div>
        <div className='about-card'>
          <div className='image'>
            <img src={aboutThree} alt='analytics´' />
          </div>
          <div className='card-info'>
            <h6>Get Hired</h6>
            <p>Ace the real interview and land that dream job</p>
          </div>
        </div>
      </div>
    </div>

    <div className='user-stories-container'>
      <h2 className='user-stories-title'>
        Hear more about how DevCoach helped students master the
        technical interview and land their dream job
      </h2>

      <div className='user-stories'>
        <div className='user-story'>
          <div className='user-header'>
            <img src={personOne} alt='user' />
            <h3>John Davidson</h3>
          </div>
          <p>
            DevCoach definitely played a role in my performance.
            Nothing beats mock coding interviews. I still remember my
            first coding interview a few years back with another
            company.
          </p>
        </div>
        <div className='user-story'>
          <div className='user-header'>
            <img src={personTwo} alt='user' />
            <h3>Aaron Ruiz</h3>
          </div>
          <p>
            Practicing with DevCoach gave me the confidence I needed
            to have a fun coding interview and eventually get the
            offers I wanted. Believe it or not, this time I wasn't
            even nervous when the day came
          </p>
        </div>
        <div className='user-story'>
          <div className='user-header'>
            <img src={personThree} alt='user' />
            <h3>Jessica Booth</h3>
          </div>
          <p>
            DevCoach helped me to get an amazing hands-on technical
            interview experience. After some preparation, I decided to
            test my skills in algorithms, so I scheduled a couple of
            DevCoach interviews.
          </p>
        </div>
      </div>
    </div>
  </MainContainer>
);

export default Main;
