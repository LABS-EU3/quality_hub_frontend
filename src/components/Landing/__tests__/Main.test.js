import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import * as rtl from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Main from '../Main/Main';

describe('Main page', () => {
  it('matches snapshot ', () => {
    const tree = renderer
      .create(
        <Router>
          <Main />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Main />
      </Router>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Main about top ok', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByTestId('main-about-top');
    expect(test).toBeInTheDocument();
    expect(test).toHaveClass('main-about-top');
  });

  it('Main h2 text ok', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.queryAllByText('Enhance');
    expect.arrayContaining(test);
  });

  it('DevCoach ok', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const text = tree.queryByText('DevCoach');
    expect(text).toBeInTheDocument();
  });

  it('Main-about-title ok', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByTestId('main-about-title');
    expect(test).toBeInTheDocument();
    expect(test).toHaveClass('main-about-title');
  });

  it('Image AboutOne in doc', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByTestId('aboutOne');
    expect(test).toBeInTheDocument();
  });

  it('Image AboutTwo in doc', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByAltText(/hand/i);
    expect(test).toBeInTheDocument();
  });

  it('Image AboutThree in doc', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByTestId('aboutThree');
    expect(test).toBeInTheDocument();
  });

  it('Image PersonOne in doc', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByTestId('personOne');
    expect(test).toBeInTheDocument();
  });

  it('Image PersonTwo in doc', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByTestId('personTwo');
    expect(test).toBeInTheDocument();
  });

  it('Image PersonThree in doc', () => {
    const tree = rtl.render(
      <Router>
        <Main />
      </Router>,
    );
    const test = tree.getByTestId('personThree');
    expect(test).toBeInTheDocument();
  });
});
