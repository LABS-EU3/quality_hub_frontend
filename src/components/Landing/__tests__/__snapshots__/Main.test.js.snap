// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page matches snapshot  1`] = `
<div
  className="sc-bdVaJa gkSaTB"
>
  <div
    className="main-about-top"
    data-testid="main-about-top"
  >
    <h2
      className="main-about-title"
      data-testid="main-about-title"
    >
      Enhance your skills. Land your dream job.
      <span
        className="logo"
      >
         DevCoach
      </span>
    </h2>
    <p
      className="main-about-intro"
    >
      We connect developers looking to improve their interviewing technique with experienced pros who have mastered the technical interview and can coach you through the process from start to finish.
    </p>
  </div>
  <div
    className="main-about-bottom"
  >
    <h4>
      How does DevCoach work?
    </h4>
    <div
      className="about-cards"
    >
      <div
        className="about-card"
      >
        <div
          className="image"
        >
          <img
            alt="hand shake´"
            src="how-2.jpg"
          />
        </div>
        <div
          className="card-info"
        >
          <h6>
            Find Your Coach
          </h6>
          <p>
            Match with experienced professionals tailored to you
          </p>
        </div>
      </div>
      <div
        className="about-card"
      >
        <div
          className="image"
        >
          <img
            alt="analytics´"
            data-testid="aboutOne"
            src="how-1.jpg"
          />
        </div>
        <div
          className="card-info"
        >
          <h6>
            Enhance Your Skills
          </h6>
          <p>
            Practice mock interviews one on one with your coach
          </p>
        </div>
      </div>
      <div
        className="about-card"
      >
        <div
          className="image"
        >
          <img
            alt="analytics´"
            data-testid="aboutThree"
            src="how-3.jpg"
          />
        </div>
        <div
          className="card-info"
        >
          <h6>
            Get Hired
          </h6>
          <p>
            Ace the real interview and land that dream job
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="user-stories-container"
  >
    <h2
      className="user-stories-title"
    >
      Hear more about how DevCoach helped students master the technical interview and land their dream job
    </h2>
    <div
      className="user-stories"
    >
      <div
        className="user-story"
      >
        <div
          className="user-header"
        >
          <img
            alt="user"
            data-testid="personOne"
            src="person-1.jpg"
          />
          <h3>
            John Davidson
          </h3>
        </div>
        <p>
          DevCoach definitely played a role in my performance. Nothing beats mock coding interviews. I still remember my first coding interview a few years back with another company.
        </p>
      </div>
      <div
        className="user-story"
      >
        <div
          className="user-header"
        >
          <img
            alt="user"
            data-testid="personTwo"
            src="person-2.jpg"
          />
          <h3>
            Aaron Ruiz
          </h3>
        </div>
        <p>
          Practicing with DevCoach gave me the confidence I needed to have a fun coding interview and eventually get the offers I wanted. Believe it or not, this time I wasn't even nervous when the day came
        </p>
      </div>
      <div
        className="user-story"
      >
        <div
          className="user-header"
        >
          <img
            alt="user"
            data-testid="personThree"
            src="person-3.jpg"
          />
          <h3>
            Jessica Booth
          </h3>
        </div>
        <p>
          DevCoach helped me to get an amazing hands-on technical interview experience. After some preparation, I decided to test my skills in algorithms, so I scheduled a couple of DevCoach interviews.
        </p>
      </div>
    </div>
  </div>
</div>
`;
